src/main/java/core/mate/academy/service/impl/TruckProducerImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.mate.academy.service.impl;
+
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducerImpl implements MachineProducer<Truck> {
+    private List<Truck> trucks;
```

Comment:
The `trucks` field is package-private, which could lead to unintended access from other classes within the same package. It's recommended to use a private access modifier to encapsulate the field properly.